idp/openid: Automatically register OpenID-Connect users

mhilton opened the pull request:
Where ID tokens supply a preferred username automatically create the
user without requiring completion of the registration form where
possible. This also adds testing for the openid-connect IDP; and
prefills the full name and emails fields on the registration form where
possible.

jujugui commented:
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/candid/68/

mhilton commented:
:shipit:

jujugui commented:
Status: merge request accepted. Url: http://ci-gce.theblues.io:8080/job/candid-merge

mhilton committed:
idp/openid: Automatically register OpenID-Connect users

Where ID tokens supply a preferred username automatically create the
user without requiring completion of the registration form where
possible. This also adds testing for the openid-connect IDP; and
prefills the full name and emails fields on the registration form where
possible.

mhilton committed:
update quicktest

mhilton committed:
update required go version

alesstimec commented on idp/openid/openid-connect.go:
```diff
@@ -47,6 +47,25 @@ func init() {
 	})
 }
 
+// An IdentityCreator is used to create a candid identity from the
```
nitpick: s/An IdentityCreator/IdentityCreator/

mhilton replied:
The convention for documenting types is to start the comment with an indefinite article (e.g. https://golang.org/pkg/net/http/#Request)

alesstimec commented on idp/openid/openid-connect.go:
```diff
@@ -223,50 +248,72 @@ func (idp *openidConnectIdentityProvider) callback(ctx context.Context, w http.R
+		if (upd != store.Update{}) {
+			err = idp.initParams.Store.UpdateIdentity(ctx, &existingUser, upd)
+		}
+		if err == nil {
```
nitpick: this is a bit confusing as the only way to the err to be not nil comes from line 277..

mhilton replied:
yes but line #277 might not have been attempted, and if it was, and failed, we want to carry on with the function.
